src/MongoDB.Driver.Core/Core/Clusters/ClusterDescription.cs

```diff
@@ -242,23 +275,32 @@ public ClusterDescription WithType(ClusterType value)
         }
 
         // private methods
-        private TimeSpan? CalculateLogicalSessionTimeout(IEnumerable<ServerDescription> servers)
+        private bool? CalculateAreSessionsSupported(List<ServerDescription> servers)
```

Comment:
This is the central one and only place that calculates whether sessions are supported.